src/agent/src/rpc.rs

```diff
@@ -1906,6 +1917,91 @@ mod tests {
         assert!(result.is_err(), "expected add arp neighbors to fail");
     }
 
+    #[tokio::test]
+    async fn test_get_memory_info() {
+        #[derive(Debug)]
+        struct TestData<'a> {
+            // if None is provided, no files will be generated, else the data in the Option will populate the file
+            block_size_data: Option<&'a str>,
+            hotplug_probe_data: Option<&'a str>,
+
+            result: (u64, bool),
+            error_contains: &'a str,
+        }
+
+        let tests = &[
+            TestData {
+                block_size_data: Some("10000000"),
+                hotplug_probe_data: Some("hotplug data"),
+                result: (268435456, true),
+                error_contains: "",
+            },
+            TestData {
+                block_size_data: Some("100"),
+                hotplug_probe_data: None,
+                result: (256, false),
+                error_contains: "",
+            },
+            TestData {
+                block_size_data: None,
+                hotplug_probe_data: None,
+                result: (0, false),
+                error_contains: "",
+            },
+            TestData {
+                block_size_data: Some(""),
+                hotplug_probe_data: None,
+                result: (0, false),
+                error_contains: "Invalid block size",
+            },
+            TestData {
+                block_size_data: Some("-1"),
+                hotplug_probe_data: None,
+                result: (0, false),
+                error_contains: "Invalid block size",
+            },
+            TestData {
+                block_size_data: Some("    "),
+                hotplug_probe_data: None,
+                result: (0, false),
+                error_contains: "Invalid block size",
+            },
+        ];
```

Comment:
It seems we're not exercising the `Err(e)` path of the function.

Maybe creating or not the block_size_path / hotplug_probe_path as part of the TestData would be an way to easily be able to check the rainy path as well.

Also, are we fully exercising the hotplug path? It seems we're focusing more on the block_size part of the function.

Reply:
> Maybe creating or not the block_size_path / hotplug_probe_path as part of the TestData would be an way to easily be able to check the rainy path as well.

I might be misunderstanding, but currently both `block_size_data` and `hotplug_probe_data` are of the type `Option`. In the `TestData` struct, if they are set to `None`, the file is not created. Is that what you were looking for?

> Also, are we fully exercising the hotplug path? It seems we're focusing more on the block_size part of the function.

The hotplug path "file not existing" path is being handled, but the path not currently being handled in both the block size and hotplug paths is if an error other than "does not exist" occurs when reading the file. The best idea I had to test that was changing permissions of the file to deny reading, but I wasn't sure if that was a good idea.